### ISSUE TITLE
Error: Member not found. (0x80020003) when accessing a COM object from javascript

### ISSUE THREAD

**Issue by david-sepulvado** [MicrosoftEdge/WebView2Feedback#1920]
**Description**
Member not found error when accessing any properties or methods from a COM object, even though others work fine.

Error message:
Uncaught Error: Member not found. (0x80020003)
    at RemoteMessenger.postSyncRequestMessage (&lt;anonymous>:1:10533)
    at Function.getHostProperty (&lt;anonymous>:1:23331)
    at Function._getAmbiguous (&lt;anonymous>:1:13956)
    at Object.get (&lt;anonymous>:1:15574)
    at &lt;anonymous>:1:3

Code:
```
// throws error
chrome.webview.hostObjects.sync.unity.ShellSecurity.Stream.UserID

// also throws error
var s = chrome.webview.hostObjects.sync.unity.ShellSecurity.Stream;
s.UserID
```

There are numerous other objects, methods, properties off the unity object that all work fine.  There is nothing special about the object's IDL, or the C++ class that implements it.  It uses the same IDispatch implementation as all of the other objects.

When I debug the C++ code for the object it is not calling GetIDsOfNames, but instead calls Invoke with a DISPID of zero and a parameter of "UserID".  The other objects that work fine first call GetIDsOfNames and then invoke with the returned DISPID.

**Version**
SDK: 1.0.818.41
Runtime: 95.0.1020.44
Framework: WPF
OS: Win10 20H2 19042.1348

**Repro Steps**

**Screenshots**

**Additional context**
IDL Excerpt:
```
[uuid(...), oleautomation, dual]
interface ISecurity : IDispatch
{
  [id(1), helpstring("method Deserialize")]
  HRESULT Deserialize([in] BSTR EncodedString);

  [id(2), propget, helpstring("property UserID")]
  HRESULT UserID([out, retval] long * UserID);

  ...
}
```

**Comment by champnic** [comment]
Hey @david-sepulvado - Do other properties/methods work off of the Stream object?

@david-risney Anything jump out to you?

**Comment by david-sepulvado** [comment]
Yes, the Stream object works fine.  I have also tried other object graphs with similar depth and they work as well.

**Comment by pontusn** [comment]
Can you share the entire IDL? Which implementation of IDispatch are you using? 

I remember dealing with something similar in our rich and complex COM API. Need more context to help out.

We now have a customized WebView2-friendly IDispatch-implementation derived from ATL to handle (few) remaining compatibility problems.

**Comment by david-risney** [comment]
Nothing in particular jumps out. Doing an Invoke for DISPID 0 is for doing a get/set/invoke on the object itself. So if an IDispatch represented a callback method, you could Invoke with DISPID 0 to run that callback method. I'm not sure why we'd do DISPID 0 with a parameter of the property name.

When you say other properties work fine, are those properties also getters for a basic type like long or int and not an object?

**Comment by pontusn** [comment]
My guess is that the bridge between JS and COM makes some assumptions about DISPID_VALUE.

Since our COM APIs must stay compatibile we cannot change DISPID, but the actual IDispatch implementation leaves plenty of root for fixes and workarounds. 

The true challenges are in handling COM DATE, which by default are interpreted as Number in JS.

**Comment by pontusn** [comment]
Maybe time to review my workarounds for WebView2 and report some more issues just to simply help others. Our integration runs fine with the fixes in place...

**Comment by pontusn** [comment]
I did some additional searching in my coding backlog...

About one year ago we had very similar issues for an API using DISPID_VALUE, while other very similar APIs using other ids worked flawless. This eventually load us to the conclusion that DISPID_VALUE cannot be returned for members exposed to WebView2.

Workaround:
Update implementation of IDispatch to avoid exposing DISPID_VALUE to caller.

Fix:
Update COM/JS-bridge in WebView2 to properly handle DISPID_VALUE.

@david-risney 
@champnic

**Comment by pontusn** [comment]
#663 also appears to describe similar issue.

**Comment by david-sepulvado** [comment]
Are there any plans to resolve the issues with DISPID_VALUE in the COM/JS bridge?

@pontusn was correct, it was the Stream property not the object itself.  The Stream property had dispid DISPID_VALUE, so whatever JS proxy was returned for that object was in a bad state.  I was able to expose the Stream property as Stream2 with a different dispid and it started working.  Note, this behavior is different for scalar properties using DISPID_VALUE.  For those the member not found error is reported immediately on property access.

Note, for my purposes, I'm not really interested in the 'default' property behavior, just being able to access those properties by name.

**Comment by champnic** [comment]
I've updated the title based on the cause and added this as a bug on our backlog to get looked at. Thanks!
